### ISSUE TITLE
Complicated sources cause Companion GUI glitches

### ISSUE THREAD

**Issue by mbr0wn** [gnuradio/gnuradio#960]
tl;dr osmocom source and UHD source blocks are pretty complex and the GUI takes noticeable time to crunch through it on every click

Hi,
when you add osmocom source from gr-osmosdr into the flowgraph, you can notice that every flowgraph change (adding connection, editing block settings...) triggers an annoying lag with 100% CPU usage.

This started with commit aa4094b3fd in gr-osmosdr which increased number of channels in the source block. Now this settings is in grc/gen_osmosdr_blocks.py:448.

UHD source has similar problem, though it takes a little shorter, so it's not as annoying.

I have measured it on i3-2350M for several numbers of channels.

Channels      Lag
5             none
32 (default)  noticeable (~500 ms?)
128           3s
256           25s

As you can see it increases more than linearly though the XML description of the block increases only linearly.

I don't know how to profile this. The workaround is setting maximum number of channels back to 5 (if you don't have radio with more channels).

**Comment by marcusmueller** [comment]
Can't reproduce – GRC is pretty instant when I change N_channels from 1 to 9 on the USRP Source. Probably got fixed on the way.
